neural/neural_discern_run_script.py

```diff
@@ -0,0 +1,327 @@
+
+    return micro_f1_df, macro_f1_df, accuracy_df
+
+
+def highlight_attnw_over_sents(docid_attnweights_map, proc_articles_repr, topk=5):
+    for docid in docid_attnweights_map:
+        attnw = docid_attnweights_map[docid]
+        topk = topk if attnw.size(-1) > topk else attnw.size(-1)  # get top
+        max_val, max_indx = torch.topk(attnw, topk, dim=1)
+        print(docid)
+        print("attended sent:")
+        for i in range(max_indx.size(-1)):
+            target_indx = max_indx[0][i].item()
+            print("sentence num:", target_indx, "attnw:", max_val[0][i].item())
+            print(proc_articles_repr[docid]['sents'][target_indx])
+        print()
+
+
+def verbose_print(text, print_yn):
+    if print_yn:
+        print(text)
+
+# ============================================================
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--test-mode", action="store_true", default=False, help="run in abbreviated test mode")
+    parser.add_argument("--biobert", default=True, help="Use the BioBERT embeddings, or the normal BERT")
+    parser.add_argument("--rewrite-sentence-embeddings", action="store_true", default=False, help="(Re-)compute the "
+                                                                                                  "sentence embeddings")
+    parser.add_argument("--run-hyper-param-search", default=True, help="Run the hyper parameter search")
+    args = parser.parse_args()
+
+    config = {
+        'test_mode': args.test_mode,
+        'biobert': args.biobert,
+        'rewrite_sentence_embeddings': args.rewrite_sentence_embeddings,
+        'run_hyper_param_search': args.run_hyper_param_search,
+        'questions_to_run': [4, 5, 9, 10, 11],
+        'max_folds': 5,
+        'num_epochs': 25,
+        'verbose': True,
+        'experiment_to_rerun': None,
+        'questions': (4, 5, 9, 10, 11),
+        'question_gpu_map': {4: 1, 5: 2, 9: 3, 10: 4, 11: 5},
+    }
+
+    if config['test_mode']:
+        config['max_folds'] = 2
+        config['num_epochs'] = 2
+        config['run_hyper_param_search'] = False
+
+    # print config to screen
+    print("{0} RUNNING WITH CONFIG {0}".format('='*10))
+    for c in config:
+        print("{}: {}".format(c, config[c]))
+    print("{0}".format('='*10*4))
+
+    base_dir = '/opt/data/autodiscern/aa_neural'
+    if config['experiment_to_rerun'] is None:
+        time_stamp = datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+        exp_dir = os.path.join(base_dir, 'experiments', time_stamp)
+        create_directory(exp_dir)
+    else:
+        exp_dir = os.path.join(base_dir, 'experiments', config['experiment_to_rerun'])
+
+    if config['biobert']:
+        bert_model_dir = '/opt/data/autodiscern/aa_neural/pytorch_biobert'
+        sents_embed_dir_name = 'sents_bert_embed_cased'
+    else:
+        bert_model_dir = '/opt/data/autodiscern/aa_neural/aws_downloads/bert-base-uncased.tar.gz'
```

Comment:
I think making all these links reconstructed based on the `base_dir` is better option.